### PR TITLE
Add a simple command line tool to test files against a schema

### PR DESCRIPTION
**Pull request by jswelling**
This tool checks .yml or .json files against the schemata in the src/ingest-pipeline/schemata directory , using the jsonschema support routines in the commons library.

### CODE REVIEW

File: src/ingest-pipeline/misc/tools/schema_tester.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+)
+from hubmap_commons.schema_tools import (
+    check_json_matches_schema
+)
+
+JSONType = Union[str, int, float, bool, None, Dict[str, Any], List[Any]]
+
+SCHEMA_BASE_PATH = join(dirname(dirname(dirname(realpath(__file__)))),
+                        'schemata')
+SCHEMA_BASE_URI = 'http://schemata.hubmapconsortium.org/'
```
Comment (sunset666) [inline src/ingest-pipeline/misc/tools/schema_tester.py:34]: Is this in use? I haven't been able to reach such url.

File: src/ingest-pipeline/misc/tools/schema_tester.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+def file_to_json(fname):
+    if fname.lower().endswith('.json') or fname.lower().endswith('.jsn'):
+        with open(fname, "r") as f:
+            tjson = json.load(f)
+    else:
+        assert fname.lower().endswith('.yaml') or fname.lower().endswith('.yml'), \
+            "File type of %s is not understood" % fname
+        with open(fname, "r") as f:
+            tjson = yaml.safe_load(f)
+            # print tjson
```
Comment (sunset666) [inline src/ingest-pipeline/misc/tools/schema_tester.py:46]: Please remove commented code

**Review by sunset666 — CHANGES_REQUESTED** [review]

File: src/ingest-pipeline/misc/tools/schema_tester.py (hunk 1 above)
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/schema_tester.py:34]: This is a strange bit- it's syntactically necessary to include a URL in the jsonschema, but it doesn't have to be a *real* URL, just a unique one.  I think it's serving as an anchor for parsing the rest of the schema reference.  But no, there is no such URL in real life.

File: src/ingest-pipeline/misc/tools/schema_tester.py (hunk 2 above)
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/schema_tester.py]: Done.

**Review by sunset666 — APPROVED** [review]

**Pull request merged**